src/k8s/pkg/k8sd/controllers/csrsigning/controller.go

```diff
@@ -92,7 +95,14 @@ func (c *Controller) Run(ctx context.Context, getClusterConfig func(context.Cont
 		Logger:  mgr.GetLogger(),
 		Client:  mgr.GetClient(),
 
-		getClusterConfig: getClusterConfig,
+		getClusterConfig:       getClusterConfig,
+		validateCSR:            validateCSR,
+		loadCertificateRequest: pkiutil.LoadCertificateRequest,
+		generateSerialNumber:   pkiutil.GenerateSerialNumber,
+		loadCertificate:        pkiutil.LoadCertificate,
+		createCertificate:      x509.CreateCertificate,
+		encodeToMemory:         pem.EncodeToMemory,
+		nowFn:                  time.Now,
```

Comment:
thanks for the comment @neoaggelos. I tried to change the tests to test the behavior rather than code paths. I completely agree on not to follow the code paths in general, but here it feels like there are a lot of things to worry about. specially that we have different things happening for different signer names (kubelet-serving, kubelet-client, ...). I agree that the initial tests that I added were really brittle, but I feel like this controller itself is a bit too brittle. anyways, changed the tests, lemme know if they need further improvements.